### PR TITLE
Add row locking, fix sidebar menu font, inbox

### PR DESCRIPTION
**Pull request by aumu1031**
- Add functions for start/commit/rollback transaction to System
- Add start transaction to all entry after System start
- Add commit to all exit
- Add rollback to all error
- Include System variable to toggle row locking
- Avoid touching specific files for now (register, password, paypal)

### CODE REVIEW

File: api/legacy_special_missions.php [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
 $system = new System();
+$system->startTransaction();
 $system->is_api_request = true;
 
 try {
     $player = Auth::getUserFromSession($system);
     $player->loadData();
 } catch(Exception $e) {
     echo json_encode(['logout' => true]);
+    $system->rollbackTransaction();
```
Comment (levimeahan) [inline api/legacy_special_missions.php:17]:
```suggestion
    $system->rollbackTransaction();
    error_log($e->getMessage());
```

We should capture these errors in server logs so we can see them

File: api/notification.php [hunk 2]
```diff
@@ -4,14 +4,15 @@
 require_once __DIR__ . "/../classes.php";
 
 $system = new System();
+$system->startTransaction();
```
Comment (levimeahan) [inline api/notification.php:7]: Huh, wonder if we should just add this call to the constructor method of System (with a parameter to disable it if desired)

File: classes/System.php [hunk 3]
```diff
@@ -340,6 +343,27 @@ public function db_fetch_all($result = false, ?string $id_column = null): array
...
 
+    public function startTransaction() {
+        if ($this->ENABLE_ROW_LOCKING) {
+            $this->query("START TRANSACTION;");
+            $this->in_db_transaction = true;
+        }
+    }
+
+    public function commitTransaction() {
+        if ($this->ENABLE_ROW_LOCKING) {
```
Comment (levimeahan) [inline classes/System.php:354]:
```suggestion
        if ($this->ENABLE_ROW_LOCKING && $this->in_db_transaction) {
```

File: classes/System.php [hunk 4]
```diff
@@ -340,6 +343,27 @@ public function db_fetch_all($result = false, ?string $id_column = null): array
...
+            $this->query("COMMIT;");
+            $this->in_db_transaction = false;
+        }
+    }
+    public function rollbackTransaction()
+    {
+        if ($this->ENABLE_ROW_LOCKING) {
```
Comment (levimeahan) [inline classes/System.php:361]:
```suggestion
        if ($this->ENABLE_ROW_LOCKING && $this->in_db_transaction) {
```

File: manual.php [hunk 5]
```diff
@@ -3,6 +3,7 @@
 
 require "classes/System.php";
 $system = new System();
+$system->rollbackTransaction();
```
Comment (levimeahan) [inline manual.php:6]:
```suggestion
$system->startTransaction();
```

File: index.php [hunk 6]
```diff
@@ -110,6 +111,7 @@
 				throw new Exception("Invalid password! <a href='./password_reset.php'>Forgot password?</a>");
 			}
 		} catch (Exception $e) {
+            $system->rollbackTransaction();
 			$system->message($e->getMessage());
```
Comment (levimeahan) [inline index.php:115]:
```suggestion
			$system->message($e->getMessage());
			error_log($e->getMessage());
```

Comment (levimeahan) [reply index.php:115]: Wonder if we should roll this into one system method, like we have `$system->error` right now, maybe it works or maybe we make a new method that just calls the three lines

```php
$system->rolbackTransaction();
error_log($error_message);
$system->message($error_message);
```

File: classes/API.php [hunk 7]
```diff
@@ -7,7 +7,8 @@
 class API {
 
     #[NoReturn]
-    public static function exitWithError(string $message, array $debug_messages = []): void {
+    public static function exitWithError(string $message, array $debug_messages = [], System $system): void {
+        $system->rollbackTransaction();
```
Comment (levimeahan) [inline classes/API.php:11]:
```suggestion
        $system->rollbackTransaction();
        error_log($message);
```

File: support.php [hunk 8]
```diff
@@ -90,6 +91,7 @@
                 }
             }
         }catch (Exception $e) {
+            $system->rollbackTransaction();
```
Comment (levimeahan) [inline support.php:94]:
```suggestion
            $system->rollbackTransaction();
            error_log($e->getMessage());
```

File: support.php [hunk 9]
```diff
@@ -157,6 +159,7 @@
                         throw new Exception("Error adding response!");
                     }
                 } catch (Exception $e) {
+                    $system->rollbackTransaction();
```
Comment (levimeahan) [inline support.php:162]:
```suggestion
                    $system->rollbackTransaction();
                    error_log($e->getMessage());
```

File: support.php [hunk 10]
```diff
@@ -193,6 +196,7 @@
                         $system->message("Support closed.");
                     }
                 }catch (Exception $e) {
+                    $system->rollbackTransaction();
```
Comment (levimeahan) [inline support.php:199]:
```suggestion
                    $system->rollbackTransaction();
                    error_log($e->getMessage());
```

File: support.php [hunk 11]
```diff
@@ -291,6 +295,7 @@
                 $system->message("Error creating support.");
             }
         }catch(Exception $e) {
+            $system->rollbackTransaction();
```
Comment (levimeahan) [inline support.php:298]:
```suggestion
            $system->rollbackTransaction();
            error_log($e->getMessage());
```

File: api/notification.php (hunk 2 above)
Comment (aumu1031) [reply api/notification.php:7]: We might want to be explicit for visibility since we're being explicit with `$system->commit/rollbackTransaction`?

Comment (levimeahan) [reply api/notification.php:7]: Yeah, we could leave this for now. But we're pulling the commit into `API::exitWithData`. Maybe the way to go is to make an `API::init` method that returns the $system object with transaction started and `is_api_request` set to true

Comment (aumu1031) [reply api/notification.php:7]: That approach makes sense. So roll this functionality into the API class:
```
$system = new System();
$system->startTransaction();
$system->is_api_request = true;
return $system;
```
and call this at the start of each API?
`$system = API::init();`

Follow-up PR okay?

Comment (levimeahan) [reply api/notification.php:7]: Yup, sounds good!

**Pull request merged**